cmd/fakeipam/fakeipam.go

```diff
@@ -49,30 +48,43 @@ func loadIpamConfig(rawConfig []byte) (danmtypes.IpamConfig,error) {
 //TODO: CNI 0.2.0 style of result is used because SRIOV plugin can't handle newer format
 //This should be generalized though, and return result should be current.Result in most cases
 func createCniResult(ipamConf danmtypes.IpamConfig) (*types.Result,error) {
-  _, ip, err := net.ParseCIDR(ipamConf.Ip + "/" + strings.Split(ipamConf.Subnet, "/")[1])
-  if err != nil {
-    return nil, errors.New("Can't parse IP from IPAM config because:"+err.Error())
-  }
-  ip.IP = net.ParseIP(ipamConf.Ip)
-  var routes []gentypes.Route
-  for _, route := range ipamConf.Routes {
-    _, destNet, err := net.ParseCIDR(route.Dst)
-    if err == nil {
-      routes = append(routes, gentypes.Route {
-        Dst: *destNet,
-        GW: net.ParseIP(route.Gw),
+  var ip net.IP
+  var ipNet = new(net.IPNet)
+  var ip4 = new(types.IPConfig)
+  var ip6 = new(types.IPConfig)
+  var cniRoutes = []gentypes.Route{}
+  for _, ipamIp := range ipamConf.Ips {
+    ip, ipNet, _ = net.ParseCIDR(ipamIp.IpCidr)
+    cniRoutes = nil
+    for _, ipamRoute := range ipamIp.Routes {
```

Comment:
the routes related section actually could be entirely deleted. we stopped pushing routes to fakeipam, because we will provision them in a generic manner from DANM, for all backends
so currently this part is dead code

Reply:
removed